transport/nats/publisher.go

```diff
@@ -0,0 +1,138 @@
+package nats
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/go-kit/kit/endpoint"
+	"github.com/nats-io/go-nats"
+	"time"
+)
+
+// Publisher wraps a URL and provides a method that implements endpoint.Endpoint.
+type Publisher struct {
+	publisher *nats.Conn
+	subject   string
+	reply     string
+	enc       EncodeRequestFunc
+	dec       DecodeResponseFunc
+	before    []RequestFunc
+	after     []ClientResponseFunc
+	async     bool
+	timeout   time.Duration
+}
+
+// NewClient constructs a usable Publisher for a single remote method.
+func NewPublisher(
+	publisher *nats.Conn,
+	subject string,
+	reply string,
+	enc EncodeRequestFunc,
+	dec DecodeResponseFunc,
+	options ...PublisherOption,
+) *Publisher {
+	c := &Publisher{
+		publisher: publisher,
+		subject:   subject,
+		reply:     reply,
+		enc:     enc,
+		dec:     dec,
+		before:  []RequestFunc{},
```

Comment:
No need to initialize this, Go already handles nil correct for all the cases needed.